runtime/compilers/rillv1/parse_dotenv.go

```diff
@@ -2,25 +2,27 @@ package rillv1
 
 import (
 	"context"
-	"os"
 
 	"github.com/joho/godotenv"
 )
 
-// parseDotEnv parses the env file present at repo root
+// parseDotEnv parses the .env file at the given path and merges it with the existing env vars
 func (p *Parser) parseDotEnv(ctx context.Context, path string) error {
 	data, err := p.Repo.Get(ctx, path)
 	if err != nil {
-		if os.IsNotExist(err) {
-			return nil
-		}
 		return err
 	}
 	envMap, err := godotenv.Unmarshal(data)
 	if err != nil {
 		return err
 	}
 
-	p.DotEnv = envMap
+	if p.DotEnv == nil {
+		p.DotEnv = make(DotEnvMap)
+	}
+
+	// Store the env vars for this specific path
+	p.DotEnv[path] = envMap
```

Comment:
If a `.env` file is deleted, how is it subsequently removed from here?

Is that why you removed the check for file existence above? I see how that would work, but it feels fragile. It would be better to clear the entry explicitly in `reparseExceptRillYAML` (which is responsible for clearing state that will be reloaded) and let `parsePaths` re-populate it.